### DISCUSSION TITLE
v0.4 models add/deletion feature now missing

### DISCUSSION THREAD

**Discussion by vulcan4d** [open-webui/open-webui#7054] [General]
Discussion https://github.com/open-webui/open-webui/discussions/7054

Go to 

```
Admin Panel
    ➡️ Settings
        ➡️Connections
             ➡️ Manage Ollama API connections
                 ➡️ Manage (🔧 icon) 
```
![image](https://github.com/user-attachments/assets/1aed8eb8-bc90-4b35-af70-fc8e22711152)

**Comment by dpk-it** [comment]
Manage Ollama Connections: The "Models" section in Admin Settings has been relocated to Admin Settings > "Connections" > Ollama Connections. You can now manage Ollama instances via a dedicated "Manage Ollama" modal from "Connections", streamlining the setup and configuration of Ollama models.

**Comment by tjbck** [comment]
You cannot delete "base" models, however, you can toggle the switch to disable them.

**Comment by EHRETic** [comment]
Hi there,

Though I was crazy after updating...
I often remove and/or add models and if this is still possible, it is not intuitive at all.

Please bring back some buttons, please 😉

**Reply by Nurb4000** [reply]
I totally agree its confusing now but after some poking i found the dialog we are looking for. ( now if i can figure out why only one model shows in my workspace... )

Admin panel -> settings -> connections -> little wrench on "Manage Ollama API Connections" down at the bottom, *not* the gear. ( tool tip = manage )                that is pretty much like what we are used to for pull/remove and gguf. 

Dont get me wrong,  I'm all for updating interfaces between releases to make it more friendly or add missing features, but this last release, nope.  


EDIT :  Ok i figured out the workspace issue. it only now shows prompt models, not the 'real' models.  Not fond of this change either.  In my case id normally just use the 'real' models from workspace. I only had a prompt model as i was playing around at one point in the past.   I can work around this of course, but still.....

**Reply by EHRETic** [reply]
I found it too after some pocking... but without @dpk-it I would not have been able to find it at all... 😁

**Comment by tjbck** [comment]
Related: https://github.com/open-webui/open-webui/discussions/7099

Hi everyone, thanks for sharing your feedback. I want to take some time to explain why these changes were made and the rationale behind them. I understand that they may feel unintuitive right now, but they serve some larger purposes that align with long-term usability and scalability, especially for a vast majority of our users. That said, I’m always open to constructive ideas on how we can improve – but let’s keep it civil and productive. Saying "Boo, this sucks" is neither helpful nor welcome here. If you're unhappy and unwilling to engage constructively, you're free to explore other great options in the space, like ChatGPT/Claude or other amazing self-hostable open-source UIs.

### Why Did We Make These Changes?  

These changes were made deliberately to resolve a few significant pain points and improve overall manageability, particularly in shared, multi-user, or organizational environments where granular control is absolutely essential. Here’s why:  

#### 1. **Separation of Admin Rights and Responsibilities**  
- Previously, there was no clear distinction between custom models (per user or workspace) and system-wide configurations, which led to serious management challenges for admins:  
  - Admins were forced to micromanage system-wide configurations **even when users only required workspace-specific changes**.  
  - This blurred design left organizational deployments far more prone to errors or misconfigurations.  
- By moving "Ollama Connections" into **Admin Settings > Connections > Manage Ollama**, administrators now have full control over system-wide settings in a consolidated, logical area.  
- Regular users can’t unintentionally make changes that had organizational impacts. It’s cleaner, safer, and adds proper boundary management.

#### 2. **Improved User Permissions and Flexibility**  
- Mixing custom models and base models in the same space created a **permission deadlock**:  
  - If one user added a custom model in a shared setup, it was visible to everyone, even in contexts where it wasn’t relevant. That created unnecessary clutter, added confusion, and made it practically impossible to standardize workspaces in joint environments.  
  - Workspace-specific models now **only show prompt models**, leaving "real" models for the system-wide admin panel. This strict separation ensures better alignment with permissions while still keeping flexibility for individual users.  
- This shift was especially necessary for organizations deploying Ollama setups in environments with **multiple non-technical users** – where settings changes needed control by admins, not end users.

#### 3. **Preparing for Scalability and Future Workflow Optimization**  
- One of the biggest drivers for reworking the UI was to future-proof how we handle both models and permissions for large teams or multi-user deployments. By moving away from the legacy "everything in one place" design, we've laid the groundwork for:  
  - Easier integration of **advanced permissions systems**.  
  - Streamlined workflows for both admins and users in high-scale environments.  
  - More clarity between **system-level integrations (like Ollama)** versus **workspace-level model customizations**.  
- More features now have clear, distinct homes in the settings hierarchy, leading to better organization and, eventually, new capabilities.  

---

### Acknowledging the Pain Points  
I hear you. For long-time users, it may seem as though the new setup is less intuitive. Whenever workflows change, there’s an adjustment curve – especially when the interface you're accustomed to starts functioning a bit differently. Here are the main issues you’ve flagged, and I’ll respond to each:  

#### 1. _“It’s confusing – hard to find specific settings!”_  
- I absolutely understand how moving things like Ollama configurations into nested menu structures introduced a learning curve. UI clarity around tools, like "gear vs wrench icons," also appears to have caused minor frustrations. This was a tradeoff in consolidating like-settings together for admins.  

**To Do:** I’ll prioritize iterative UI enhancements to improve visibility and ease navigation, such as clearer tool tips or highlighting newly introduced changes after updates (e.g., with overlays or system notifications). Suggestions for how you'd like this visualized are welcome.  

#### 2. _“Why can’t I see the ‘real’ models in the workspace?”_  
- This was an intentional design change – we’ve deliberately limited workspace visibility to **prompt models only** to maintain a logical boundary between user-owned content and administratively managed configurations.  

**Workaround:** If you prefer direct control over system-level models, that's still accessible through Admin Settings, provided you’re authorized to make such changes.  

#### 3. _“Bring back the buttons – it’s not user-friendly!”_  
- Context matters here. In an organizational setup, having overly prominent delete or configuration buttons could lead to accidental misuse or confusion among users unfamiliar with what each model is for. That said, I **understand the frustration in an individual use case** where this feels like unnecessary friction.  

**To Do:** I’m happy to look for ways to make access faster (e.g., a direct shortcut to “Manage Ollama” in key spots) while preserving important guardrails for larger deployments.  

---

Additionally, I’d like to remind everyone this project is **open source**, which comes with certain freedoms. If you disagree with our vision or these changes, you’re welcome to maintain your own forked version. That way, you retain control over your preferred workflows without being subject to updates you’re unhappy with. If you’re unsure how to fork, I’d be happy to point you to resources that can help.  

However, please keep this discussion productive. Saying "Boo, this sucks" without offering constructive alternatives isn’t an effective use of anyone’s time. While I welcome respectful disagreements, baseless complaints are anti-collaborative and won't help move us forward.

---

### Constructive Feedback is Welcome  

I know these changes may feel disruptive now, but they’re designed to make the tool better for a broader range of users, especially teams and organizations that rely on robust permissions and clean boundaries. If you have specific suggestions for further improvement, please share!  

Thanks as always for your passion and involvement in this project. Your feedback is instrumental to making it the best it can be – but let's keep it grounded in constructive dialogue. Appreciate your understanding, and I’ll continue working toward refinements that serve everyone.  

Best,  
Tim

**Reply by dpk-it** [reply]
I think the problem is that it is not at all obvious that adding and deleting models happens in a modal window hidden behind a small icon. It would be much better to make this button bigger and write something clear on it (like "Model Management"), and not just use an icon... or maybe it is worth writing a hint next to the list of Ollama connections telling where to find these options.

**Comment by Nurb4000** [comment]
@tjbck 
Thanks for the clarification of the reasoning. 

On the move, when i write things that need admin areas too, while do i understand your reasoning, myself i rely on roles and hide options based on that, instead of separating bits and pieces and moving them to other areas that 'make sense to me'. This way the interface layout is more consistent across users. For users "we all go to the same place, but some of us see extra options" ( sometimes even new menus, but they still go to the same place ).  But i do also understand it adds a bit of code complexity.  And yes there is some risk of accidents on rights if its missed,  but for where im at, it also helps other dev guys out "all of these 'kinds of things' are all in one spot" instead of "ok, where was this dialog again"  ( and if all functions are APIs calls, which we are moving towards those are locked down as well, so even if they DID see a button.. it wont actually do anything )

On base models vanishing from the main view, I bet that does not effect a lot of people, not many of us only used the base and not the prompt models.  I was lucky i had one prompt out there i was messing with long ago, so i knew the system wasn't broke, as that was a change id not expect.  While confused at first i did find at least 2 ways to do what i need ( once the prompt was selected i can change the base at the top, so i created a dummy one.  If that goes away as part of the 'cleanup', i can just create dummy ones for each model, instead of going to 'another area' to access them. ) . And i do understand trying to clean things up between user classes/external connections etc.  Id just have gone with the roles again, instead of moving  :)
